hydra/controllers/client.py

```diff
@@ -816,3 +817,74 @@ def disable_metrics(self):
 
         self.restart_service()
         self.app.log.info(f'Successfully disabled all metrics reporting.')
+
+    @ex(
+        arguments=[
+            (
+                ['-n', '--name'],
+                {
+                    'help': 'name of network to join',
+                    'action': 'store',
+                    'dest': 'name'
+                }
+            ),
+            (
+                ['-v', '--version'],
+                {
+                    'help': 'version of binary',
+                    'action': 'store',
+                    'dest': 'version'
+                }
+            ),
+            (
+                ['-D', '--destroy'],
+                {
+                    'help': 'destroy existing directory',
+                    'action': 'store_true',
+                    'dest': 'destroy'
+                }
+            ),
+        ]
+    )
+    def upgrade_binary(self, version=None, destroy=False):
+        version = self.app.pargs.version
+
+        p = Prompt(f'Upgrading to ShipChain binary version: {version}. Please confirm (y/n):')
```

Comment:
I think it'd be a good idea to include the network name in this message, just so the user can confirm exactly what's going on.

Reply:
SGTM